plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/simulate/SimulatedItemResponse.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.plugin.ingest.simulate;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.xcontent.StatusToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentBuilderString;
+import org.elasticsearch.ingest.Data;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class SimulatedItemResponse implements Streamable, StatusToXContent {
+
+    private Data data;
+    private List<ProcessedData> processedDataList;
+    private Throwable failure;
+
+    public SimulatedItemResponse() {
+
+    }
+
+    public SimulatedItemResponse(Data data) {
+        this.data = data;
+    }
+
+    public SimulatedItemResponse(List<ProcessedData> processedDataList) {
+        this.processedDataList = processedDataList;
+    }
+
+    public SimulatedItemResponse(Throwable failure) {
+        this.failure = failure;
+    }
+
+    public boolean isFailed() {
+        return this.failure != null;
+    }
+
+    public boolean isVerbose() {
+        return this.processedDataList != null;
+    }
+
+    public Data getData() {
+        return data;
+    }
+
+    public List<ProcessedData> getProcessedDataList() {
+        return processedDataList;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        boolean isFailed = in.readBoolean();
+        boolean isVerbose = in.readBoolean();
+        if (isFailed) {
+            this.failure = in.readThrowable();
+            // TODO(talevy): check out mget for throwable limitations
```

Comment:
this todo can be removed. This now also how mget serialises errors. Before this was different.